builder/amazon/ebssurrogate/step_register_ami.go

```diff
@@ -25,6 +25,8 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 
 	ui.Say("Registering the AMI...")
 
+	s.BlockDevices = config.BlockDevices.BuildAMIDevices()
```

Comment:
:+1:, thats the correct place to put this, at which point I think this is valid and will fix the issue.